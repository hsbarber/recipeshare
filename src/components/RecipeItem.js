import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import AnimateHeight from 'react-animate-height';
import ChevronDown from '../icons/ChevronDown';

function RecipeItem(props) {
  const { toggleModal, removeID, removeRecipe, recipe, user } = props;
  // console.log(recipe);
  return (
    <li key={recipe.id}>
      <Link to={`/${recipe.title.split(' ').join('')}`}>
        {recipe.imageURL ? (
          <div
            className="bgImage"
            style={{
              backgroundImage: `url(${recipe.imageURL})`,
            }}
          />
        ) : (
          <div
            className="bgImage"
            style={{
              backgroundColor: `#627bc0`,
            }}
          >
            <h4>No Image Available</h4>
          </div>
        )}
        <h3 className="category">{recipe.category}</h3>
        <div className="title-block">
          <h3 className="title">{recipe.title}</h3>
          {recipe.user && (
            <h4>
              by<span>{recipe.user}</span>
            </h4>
          )}
        </div>
      </Link>
      {recipe.user
        ? recipe.user === user && (
            <button
              className="remove"
              type="button"
              onClick={() => toggleModal(recipe.id)}
            >
              Remove Recipe{' '}
              {removeID[recipe.id] ? (
                <span className="down">
                  <ChevronDown color="#273762" />
                </span>
              ) : (
                <span className="up">
                  <ChevronDown color="#273762" />
                </span>
              )}{' '}
            </button>
          )
        : ''}

      {removeID[recipe.id] && (
        <AnimateHeight
          className="dropDown"
          duration={500}
          height={removeID[recipe.id] ? 'auto' : 0}
        >
          <div className="deleteDrop">
            <p>Are you sure you want to Delete?</p>
            <p className="deleteButtons">
              <button type="button" onClick={() => removeRecipe(recipe.id)}>
                Delete
              </button>
              <button
                type="button"
                className="closeModal"
                onClick={() => toggleModal(recipe.id)}
              >
                Close
              </button>
            </p>
          </div>
        </AnimateHeight>
      )}
    </li>
  );
}

export default RecipeItem;

RecipeItem.propTypes = {
  recipe: PropTypes.object,
  user: PropTypes.string,
  removeID: PropTypes.object,
  removeRecipe: PropTypes.func,
  toggleModal: PropTypes.func,
};
