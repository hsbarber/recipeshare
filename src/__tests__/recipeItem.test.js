import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import '../enzyme';
import RecipeItem from '../components/RecipeItem';

const fakeItem = {
  category: 'Mediterranean',
  email: 'haganb3@gmail.com',
  id: '-LKsBjUcn-7KrbxFIlJa',
  imageURL:
    'https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826',
  ingredients: [
    { content: 'butter', id: 1535319837489 },
    { content: 'bread', id: 1535319837488 },
    { content: 'cheese', id: 1535319837487 },
  ],
  link: undefined,
  notes: undefined,
  recipeTime: undefined,
  steps: [
    { content: 'heat up', id: 1535319837491 },
    { content: 'STIR', id: 1535319837492 },
    { content: 'serve', id: 1535319837493 },
  ],
  time: 1535319939642,
  title: 'Roasted Butternut Squash Soup',
  user: 'hagan',
};
const getDefaultProps = () => ({
  toggleModal: jest.fn(),
  removeID: {},
  removeRecipe: jest.fn(),
  user: 'hagan',
});
describe('RecipeItem', () => {
  const { toggleModal, removeID, removeRecipe, user } = getDefaultProps();
  it('shallow renders without crashing', () => {
    shallow(
      <RecipeItem
        toggleModal={toggleModal}
        removeID={removeID}
        removeRecipe={removeRecipe}
        user={user}
        recipe={fakeItem}
      />
    );
  });
  it('renders differently with requested', () => {
    const wrapper = shallow(
      <RecipeItem
        toggleModal={toggleModal}
        removeID={removeID}
        removeRecipe={removeRecipe}
        user={user}
        recipe={fakeItem}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('When user does not match recipe.user, recipe dropdown not visible', () => {
    const props = {
      toggleModal: jest.fn(),
      removeID: {},
      removeRecipe: jest.fn(),
      user: 'John',
    };
    const RecipeDropDown = shallow(
      <MemoryRouter>
        <RecipeItem {...props} recipe={fakeItem} />
      </MemoryRouter>
    );
    expect(RecipeDropDown.hasClass('remove')).toBeFalsy();
  });
});

describe('RecipeItem Toggle', () => {
  it('expect delete modal to not be visible, when removeID is blank', () => {
    const props = {
      toggleModal: jest.fn(),
      removeID: {},
      removeRecipe: jest.fn(),
      user: 'hagan',
    };
    const RecipeItemComponent = mount(
      <MemoryRouter>
        <RecipeItem {...props} recipe={fakeItem} />
      </MemoryRouter>
    );
    expect(RecipeItemComponent.find('.deleteDrop').exists()).toEqual(false);
    RecipeItemComponent.unmount();
  });
  it('delete modal becomes visible after remove button is clicked on', () => {
    const props = {
      toggleModal: jest.fn(),
      removeID: { '-LKsBjUcn-7KrbxFIlJa': true },
      removeRecipe: jest.fn(),
      user: 'hagan',
    };
    const RecipeItemComponent = mount(
      <MemoryRouter>
        <RecipeItem {...props} recipe={fakeItem} />
      </MemoryRouter>
    );

    const button = RecipeItemComponent.find('.remove');
    button.simulate('click');

    const dropDown = RecipeItemComponent.find('.dropDown');
    expect(props.toggleModal).toHaveBeenCalled();
    expect(dropDown.find('.deleteDrop').exists()).toEqual(true);

    RecipeItemComponent.unmount();
  });
});
