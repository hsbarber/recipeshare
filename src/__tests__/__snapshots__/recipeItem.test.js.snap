// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItem renders differently with requested 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeItem
    isModalOpen={false}
    recipe={
      Object {
        "category": "Mediterranean",
        "email": "haganb3@gmail.com",
        "id": "-LKsBjUcn-7KrbxFIlJa",
        "imageURL": "https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826",
        "ingredients": Array [
          Object {
            "content": "butter",
            "id": 1535319837489,
          },
          Object {
            "content": "bread",
            "id": 1535319837488,
          },
          Object {
            "content": "cheese",
            "id": 1535319837487,
          },
        ],
        "link": undefined,
        "notes": undefined,
        "recipeTime": undefined,
        "steps": Array [
          Object {
            "content": "heat up",
            "id": 1535319837491,
          },
          Object {
            "content": "STIR",
            "id": 1535319837492,
          },
          Object {
            "content": "serve",
            "id": 1535319837493,
          },
        ],
        "time": 1535319939642,
        "title": "Roasted Butternut Squash Soup",
        "user": "Frank",
      }
    }
    removeID="-LKsBjUcn-7KrbxFIlJa"
    removeRecipe={[MockFunction]}
    toggleModal={[MockFunction]}
    user="Frank"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "-LKsBjUcn-7KrbxFIlJa",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/RoastedButternutSquashSoup"
        >
          <div
            className="bgImage"
            style={
              Object {
                "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
              }
            }
          />
          <h3
            className="category"
          >
            Mediterranean
          </h3>
          <div
            className="title-block"
          >
            <h3
              className="title"
            >
              Roasted Butternut Squash Soup
            </h3>
            <h4>
              by
              <span>
                Frank
              </span>
            </h4>
          </div>
        </Link>,
        <button
          className="remove"
          onClick={[Function]}
          type="button"
        >
          Remove Recipe
           
          <span
            className="up"
          >
            <ChevronDown
              color="#273762"
            />
          </span>
           
        </button>,
        <AnimateHeight
          animateOpacity={false}
          animationStateClasses={
            Object {
              "animating": "rah-animating",
              "animatingDown": "rah-animating--down",
              "animatingToHeightAuto": "rah-animating--to-height-auto",
              "animatingToHeightSpecific": "rah-animating--to-height-specific",
              "animatingToHeightZero": "rah-animating--to-height-zero",
              "animatingUp": "rah-animating--up",
              "static": "rah-static",
              "staticHeightAuto": "rah-static--height-auto",
              "staticHeightSpecific": "rah-static--height-specific",
              "staticHeightZero": "rah-static--height-zero",
            }
          }
          applyInlineTransitions={true}
          delay={0}
          duration={500}
          easing="ease"
          height={0}
          style={Object {}}
        >
          <div
            className="deleteDrop"
          >
            <p>
              Are you sure you want to Delete?
            </p>
            <p
              className="deleteButtons"
            >
              <button
                onClick={[Function]}
                type="button"
              >
                Delete
              </button>
              <button
                className="closeModal"
                onClick={[Function]}
                type="button"
              >
                Close
              </button>
            </p>
          </div>
        </AnimateHeight>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="bgImage"
              style={
                Object {
                  "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
                }
              }
            />,
            <h3
              className="category"
            >
              Mediterranean
            </h3>,
            <div
              className="title-block"
            >
              <h3
                className="title"
              >
                Roasted Butternut Squash Soup
              </h3>
              <h4>
                by
                <span>
                  Frank
                </span>
              </h4>
            </div>,
          ],
          "replace": false,
          "to": "/RoastedButternutSquashSoup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bgImage",
              "style": Object {
                "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mediterranean",
              "className": "category",
            },
            "ref": null,
            "rendered": "Mediterranean",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="title"
                >
                  Roasted Butternut Squash Soup
                </h3>,
                <h4>
                  by
                  <span>
                    Frank
                  </span>
                </h4>,
              ],
              "className": "title-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Roasted Butternut Squash Soup",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Roasted Butternut Squash Soup",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "by",
                    <span>
                      Frank
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "by",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Frank",
                    },
                    "ref": null,
                    "rendered": "Frank",
                    "type": "span",
                  },
                ],
                "type": "h4",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Remove Recipe",
            " ",
            <span
              className="up"
            >
              <ChevronDown
                color="#273762"
              />
            </span>,
            " ",
          ],
          "className": "remove",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "Remove Recipe",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ChevronDown
                color="#273762"
              />,
              "className": "up",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#273762",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          " ",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animateOpacity": false,
          "animationStateClasses": Object {
            "animating": "rah-animating",
            "animatingDown": "rah-animating--down",
            "animatingToHeightAuto": "rah-animating--to-height-auto",
            "animatingToHeightSpecific": "rah-animating--to-height-specific",
            "animatingToHeightZero": "rah-animating--to-height-zero",
            "animatingUp": "rah-animating--up",
            "static": "rah-static",
            "staticHeightAuto": "rah-static--height-auto",
            "staticHeightSpecific": "rah-static--height-specific",
            "staticHeightZero": "rah-static--height-zero",
          },
          "applyInlineTransitions": true,
          "children": <div
            className="deleteDrop"
          >
            <p>
              Are you sure you want to Delete?
            </p>
            <p
              className="deleteButtons"
            >
              <button
                onClick={[Function]}
                type="button"
              >
                Delete
              </button>
              <button
                className="closeModal"
                onClick={[Function]}
                type="button"
              >
                Close
              </button>
            </p>
          </div>,
          "delay": 0,
          "duration": 500,
          "easing": "ease",
          "height": 0,
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Are you sure you want to Delete?
              </p>,
              <p
                className="deleteButtons"
              >
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
                <button
                  className="closeModal"
                  onClick={[Function]}
                  type="button"
                >
                  Close
                </button>
              </p>,
            ],
            "className": "deleteDrop",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Are you sure you want to Delete?",
              },
              "ref": null,
              "rendered": "Are you sure you want to Delete?",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Delete
                  </button>,
                  <button
                    className="closeModal"
                    onClick={[Function]}
                    type="button"
                  >
                    Close
                  </button>,
                ],
                "className": "deleteButtons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Close",
                    "className": "closeModal",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Close",
                  "type": "button",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "-LKsBjUcn-7KrbxFIlJa",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/RoastedButternutSquashSoup"
          >
            <div
              className="bgImage"
              style={
                Object {
                  "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
                }
              }
            />
            <h3
              className="category"
            >
              Mediterranean
            </h3>
            <div
              className="title-block"
            >
              <h3
                className="title"
              >
                Roasted Butternut Squash Soup
              </h3>
              <h4>
                by
                <span>
                  Frank
                </span>
              </h4>
            </div>
          </Link>,
          <button
            className="remove"
            onClick={[Function]}
            type="button"
          >
            Remove Recipe
             
            <span
              className="up"
            >
              <ChevronDown
                color="#273762"
              />
            </span>
             
          </button>,
          <AnimateHeight
            animateOpacity={false}
            animationStateClasses={
              Object {
                "animating": "rah-animating",
                "animatingDown": "rah-animating--down",
                "animatingToHeightAuto": "rah-animating--to-height-auto",
                "animatingToHeightSpecific": "rah-animating--to-height-specific",
                "animatingToHeightZero": "rah-animating--to-height-zero",
                "animatingUp": "rah-animating--up",
                "static": "rah-static",
                "staticHeightAuto": "rah-static--height-auto",
                "staticHeightSpecific": "rah-static--height-specific",
                "staticHeightZero": "rah-static--height-zero",
              }
            }
            applyInlineTransitions={true}
            delay={0}
            duration={500}
            easing="ease"
            height={0}
            style={Object {}}
          >
            <div
              className="deleteDrop"
            >
              <p>
                Are you sure you want to Delete?
              </p>
              <p
                className="deleteButtons"
              >
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
                <button
                  className="closeModal"
                  onClick={[Function]}
                  type="button"
                >
                  Close
                </button>
              </p>
            </div>
          </AnimateHeight>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="bgImage"
                style={
                  Object {
                    "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
                  }
                }
              />,
              <h3
                className="category"
              >
                Mediterranean
              </h3>,
              <div
                className="title-block"
              >
                <h3
                  className="title"
                >
                  Roasted Butternut Squash Soup
                </h3>
                <h4>
                  by
                  <span>
                    Frank
                  </span>
                </h4>
              </div>,
            ],
            "replace": false,
            "to": "/RoastedButternutSquashSoup",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "bgImage",
                "style": Object {
                  "backgroundImage": "url(https://firebasestorage.googleapis.com/v0/b/recipes-cc787.appspot.com/o/images%2F3a92e118-0760-4bfc-92b6-d1176a1889d4.jpg?alt=media&token=f9e3c5e5-6766-41be-a35e-d026f1075826)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mediterranean",
                "className": "category",
              },
              "ref": null,
              "rendered": "Mediterranean",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="title"
                  >
                    Roasted Butternut Squash Soup
                  </h3>,
                  <h4>
                    by
                    <span>
                      Frank
                    </span>
                  </h4>,
                ],
                "className": "title-block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Roasted Butternut Squash Soup",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Roasted Butternut Squash Soup",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "by",
                      <span>
                        Frank
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "by",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Frank",
                      },
                      "ref": null,
                      "rendered": "Frank",
                      "type": "span",
                    },
                  ],
                  "type": "h4",
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Remove Recipe",
              " ",
              <span
                className="up"
              >
                <ChevronDown
                  color="#273762"
                />
              </span>,
              " ",
            ],
            "className": "remove",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            "Remove Recipe",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ChevronDown
                  color="#273762"
                />,
                "className": "up",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": "#273762",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            " ",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animateOpacity": false,
            "animationStateClasses": Object {
              "animating": "rah-animating",
              "animatingDown": "rah-animating--down",
              "animatingToHeightAuto": "rah-animating--to-height-auto",
              "animatingToHeightSpecific": "rah-animating--to-height-specific",
              "animatingToHeightZero": "rah-animating--to-height-zero",
              "animatingUp": "rah-animating--up",
              "static": "rah-static",
              "staticHeightAuto": "rah-static--height-auto",
              "staticHeightSpecific": "rah-static--height-specific",
              "staticHeightZero": "rah-static--height-zero",
            },
            "applyInlineTransitions": true,
            "children": <div
              className="deleteDrop"
            >
              <p>
                Are you sure you want to Delete?
              </p>
              <p
                className="deleteButtons"
              >
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
                <button
                  className="closeModal"
                  onClick={[Function]}
                  type="button"
                >
                  Close
                </button>
              </p>
            </div>,
            "delay": 0,
            "duration": 500,
            "easing": "ease",
            "height": 0,
            "style": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Are you sure you want to Delete?
                </p>,
                <p
                  className="deleteButtons"
                >
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Delete
                  </button>
                  <button
                    className="closeModal"
                    onClick={[Function]}
                    type="button"
                  >
                    Close
                  </button>
                </p>,
              ],
              "className": "deleteDrop",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Are you sure you want to Delete?",
                },
                "ref": null,
                "rendered": "Are you sure you want to Delete?",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      onClick={[Function]}
                      type="button"
                    >
                      Delete
                    </button>,
                    <button
                      className="closeModal"
                      onClick={[Function]}
                      type="button"
                    >
                      Close
                    </button>,
                  ],
                  "className": "deleteButtons",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Delete",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Delete",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Close",
                      "className": "closeModal",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Close",
                    "type": "button",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
